[BUG]: PCSX2 on linux immediately aborts with assertion failed in function void recRecompile(u32): xGetPtr() - recPtr < _64kb when starting a game but not bios

### Describe the Bug

PCSX2 built from Master on Linux immediately crashes with an abort when trying to launch any game. But the bios is able to boot fine.

This has been happening for a few weeks and I've been `git pull master` a few times but the problem is still around.
I've attached the output from PCSX2 when run which includes the abort line and error line number:
`pcsx2/x86/ix86-32/iR5900-32.cpp:2675: assertion failed in function void recRecompile(u32): xGetPtr() - recPtr < _64kb`

I've also attached the cmake output from the build.
[pcsx2-config.log](https://github.com/PCSX2/pcsx2/files/11007341/pcsx2-config.log)
[pcsx2-crash.log](https://github.com/PCSX2/pcsx2/files/11007342/pcsx2-crash.log)


### Reproduction Steps

Checkout out PCSX2 git with commit `d62a7999fbdc14b56d765b80317bf82b9deb6947`
mkdir build && cd build
cmake ..
make -j24
./bin/pcsx2-qt
Double click "Crash Tag Team racing" in game list, application crashes and closes

### Expected Behavior

The game to launch and be playable

### PCSX2 Revision

v1.7.2175-3361-gd62a7999f

### Operating System

Linux (64bit) - Specify distro below

### If Linux - Specify Distro

Arch

### CPU

intel xeon x5660 * 2

### GPU

Nvidia GTX 1060 6gb driver 525.89.02

### GS Settings

Settings are attached as a screenshot
![Screenshot from 2023-03-18 14-03-49](https://user-images.githubusercontent.com/807332/226081262-f020306f-6503-43e3-888d-1cab08839b86.png)
![Screenshot from 2023-03-18 14-03-55](https://user-images.githubusercontent.com/807332/226081263-277048e8-25c1-40a1-a454-6eaf9e81a281.png)
![Screenshot from 2023-03-18 14-04-00](https://user-images.githubusercontent.com/807332/226081265-cf219d0a-49b9-4627-bdaf-1a652f9ae4e7.png)
![Screenshot from 2023-03-18 14-04-07](https://user-images.githubusercontent.com/807332/226081267-31a557ef-e08c-4ab2-a274-8b721a983a24.png)


### Emulation Settings

Settings are attached as a screenshot. Whether MTVU is enabled or not it still crashes.
![Screenshot from 2023-03-18 14-05-02](https://user-images.githubusercontent.com/807332/226081273-0cfeb44b-ff02-4285-aad2-bf275e6baeea.png)


### GS Window Screenshots

_No response_

### Logs & Dumps

_No response_

Just tested building the native linux version straight from master (NOT from AUR), PCSX2 fail to build with the following error:
```
/home/kamfretoz/stuff/pcsx2/pcsx2/GameDatabase.cpp:180:29: error: no viable conversion from 'c4::yml::ConstNodeRef' to 'const ryml::NodeRef'
for (const ryml::NodeRef& n : node["gameFixes"].children())
```
![image](https://user-images.githubusercontent.com/14798312/226116400-aa94db8b-0500-4562-ae52-96dee769f81d.png)


which seems to suggest that it fails due to `rapidyaml`

the build expect older `rapidyaml` version 0.4.1 present on the system, and will fail to build when compiled against the latest 0.5.1 version

--- a/pcsx2/x86/BaseblockEx.h
+++ b/pcsx2/x86/BaseblockEx.h
@@ -32,10 +32,10 @@ struct BASEBLOCK
 // extra block info (only valid for start of fn)
 struct BASEBLOCKEX
 {
-	u32 startpc;
 	uptr fnptr;
-	u16 size;    // The size in dwords (equivalent to the number of instructions)
-	u16 x86size; // The size in byte of the translated x86 instructions
+	u32 startpc;
+	u32 size;    // The size in dwords (equivalent to the number of instructions)
+	u32 x86size; // The size in byte of the translated x86 instructions
 
 #ifdef PCSX2_DEVBUILD
 	// Could be useful to instrument the block

--- a/pcsx2/x86/ix86-32/iR5900-32.cpp
+++ b/pcsx2/x86/ix86-32/iR5900-32.cpp
@@ -2672,8 +2672,7 @@ StartRecomp:
 
 	pxAssert(xGetPtr() < recMem->GetPtrEnd());
 
-	pxAssert(xGetPtr() - recPtr < _64kb);
-	s_pCurBlockEx->x86size = xGetPtr() - recPtr;
+	s_pCurBlockEx->x86size = static_cast<u32>(xGetPtr() - recPtr);
 
 #if 0
 	// Example: Dump both x86/EE code

